lib/livebook/hubs.ex

```diff
@@ -272,4 +273,33 @@ defmodule Livebook.Hubs do
   def capability?(hub, capabilities) do
     capabilities -- Provider.capabilities(hub) == []
   end
+
+  @offline_hub_key :livebook_offline_hubs
+
+  @doc """
+  Get the offline hubs list from persistent term.
+  """
+  @spec get_offline_hubs() :: list(Provider.t())
+  def get_offline_hubs() do
+    :persistent_term.get(@offline_hub_key, [])
+  end
+
+  @doc """
+  Sets a offline hub that will be kept only in memory.
+  """
+  @spec set_offline_hub(Provider.t()) :: :ok
+  def set_offline_hub(hub) do
+    :persistent_term.put(@offline_hub_key, [hub])
+  end
+
+  @doc """
+  Gets one hub from persistent term.
+  """
+  @spec fetch_offline_hub(String.t()) :: {:ok, Provider.t()} | :error
+  def fetch_offline_hub(id) do
```

Comment:
Why do we need a separate function for this? Shouldn't we change `fetch_hub` to also look for offline hubs?

Reply:
We did the same in the past with Secrets, but we changed later because it would more painful to deal with the startup secrets. After your review, it seems to be better to have it separated.